USF-2561: The company admin can view and edit the company profile

slavvka opened the pull request:
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--{repo}--{owner}.aem.live/
- After: https://<branch>--{repo}--{owner}.aem.live/

slavvka committed:
USF-2561: The company admin can view and edit the company profile

slavvka committed:
USF-2561: The company admin can view and edit the company profile

slavvka committed:
Merge remote-tracking branch 'mainline/integration' into USF-2561

slavvka committed:
USF-2561: The company admin can view and edit the company profile

slavvka committed:
USF-2561: The company admin can view and edit the company profile

fnhipster commented on blocks/commerce-company-profile/commerce-company-profile.js:
```diff
@@ -0,0 +1,39 @@
+import '../../scripts/initializers/company.js';
+
+export default async function decorate(block) {
+  if (!checkIsAuthenticated()) {
+    window.location.href = rootLink(CUSTOMER_LOGIN_PATH);
+    return;
```
It would be best to check for authentication at the initializer level instead of at the block level. You can refer to the order drop-in ([order.js](https://github.com/hlxsites/aem-boilerplate-commerce/blob/main/scripts/initializers/order.js)) as an example. The container should be all about rendering the content, and not be concerned about redirects during load.

That way, the check happens as soon as the drop-in is initialized rather than only when a block is used.

slavvka replied:
@fnhipster I used this approach because i saw it in storefront-account, But I agree that it makes sense to move it to the initializer.

slavvka replied:
@fnhipster I move the code as you suggested but here we have another issue. Not all blocks or other functionality like a link to create a company in the header rely on the same checks. In my case I check if a customer authenticated, if company functionality is on and if a customer has a company (and if not the page is redirected somewhere else). But for some cases like creation a company by a guest or as by a customer without a company that won't work. And we need to have this checks in blocks and in the header script separately (like it was before) or even in the dropin itself. Or we can have different initializers: one for most of the blocks and separate ones for edge cases I described. What do you think?

CC @jilu1

fnhipster replied:
@slavvka I see. Thanks for looking into this. I thought of the "authenticated" check, but that makes sense. Let's keep it simple then, and have the rules in the blocks that need them.

ishakhsuvarov commented on .gitignore:
```diff
@@ -3,6 +3,7 @@ coverage/*
 logs/*
 node_modules
 .parcel-cache
+config.json
```
Make sure this is not getting delivered.

slavvka replied:
Why?

fnhipster replied:
Started a thread in Slack about it https://magento.slack.com/archives/C03J16UAEHZ/p1757090101489599

ishakhsuvarov commented on head.html:
```diff
@@ -28,6 +28,7 @@
             "@dropins/storefront-wishlist/": "/scripts/__dropins__/storefront-wishlist/",
             "@dropins/storefront-personalization/": "/scripts/__dropins__/storefront-personalization/",
             "@dropins/storefront-product-discovery/": "/scripts/__dropins__/storefront-product-discovery/",
+            "@dropins/storefront-company-management/": "http://127.0.0.1:3002/",
```
Make sure this is not getting delivered as is

slavvka replied:
Sure

slavvka committed:
USF-2561: The company admin can view and edit the company profile

slavvka committed:
USF-2561: The company admin can view and edit the company profile

slavvka committed:
USF-2561: The company admin can view and edit the company profile

slavvka committed:
USF-2561: The company admin can view and edit the company profile

slavvka committed:
Merge remote-tracking branch 'mainline/b2b' into USF-2561

slavvka committed:
USF-2561: The company admin can view and edit the company profile

slavvka committed:
USF-2561: The company admin can view and edit the company profile

slavvka committed:
Merge remote-tracking branch 'mainline/b2b' into USF-2561

slavvka committed:
USF-2561: The company admin can view and edit the company profile

slavvka committed:
USF-2708: Support of company change event from Company Switcher

slavvka committed:
USF-2561: The company admin can view and edit the company profile

slavvka committed:
Merge branch 'USF-2561' into USF-2708

slavvka committed:
USF-2708: Support of company change event from Company Switcher

slavvka committed:
USF-2708: Support of company change event from Company Switcher

slavvka committed:
Merge pull request #6 from magento-amigos/USF-2708

[Amigos] USF-2708: Support of company change event from Company Switcher

slavvka committed:
USF-2708: Support of company change event from Company Switcher

slavvka committed:
Merge branch 'b2b' into USF-2561

jcalcaben committed:
feature: Request a negotiable quote in cart (#731)

* feat: Add b2b quote management dropin to project

* feature (b2b): Add ability to request quote from cart

* task: Fix linting issues

- disabled linting checks for imports temporarily until we have alpha packages for dropin

* task: Update package json to use published dropin as a dependency

* task: update to latest quote management package

* bugfix: Fix request quote button showing up on non company user

---------

Co-authored-by: Eric Oeur <31492142+ericeoeur@users.noreply.github.com>

slavvka committed:
Merge remote-tracking branch 'mainline/b2b' into USF-2561

slavvka committed:
Merge remote-tracking branch 'mainline/USF-2561' into USF-2561

slavvka committed:
Company Profile bug fixes
